src/System.Diagnostics.EventLog/tests/EventLogTests/EventLogTests.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Xunit;
+
+namespace System.Diagnostics.Tests
+{
+    public class EventLogTests
+    {
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void EventLogReinitializationException()
+        {
+            using (EventLog eventLog = new EventLog())
+            {
+                eventLog.BeginInit();
+                Assert.Throws<InvalidOperationException>(() => eventLog.BeginInit());
+                eventLog.Close();
+            }
+        }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void ClearLogTest()
+        {
+            if (!AdminHelpers.IsProcessElevated())
+                return;
+
+            string source = Guid.NewGuid().ToString("N");
+            EventLog.CreateEventSource(source, "MyNewLog");
+            using (EventLog myLog = new EventLog())
+            {
+                myLog.Source = source;
+                myLog.WriteEntry("Writing to event log.");
+                Assert.True(myLog.Entries.Count != 0);
+                myLog.Clear();
+                Assert.Equal(0, myLog.Entries.Count);
+            }
+        }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void ApplicationEventLog_Count()
+        {
+            EventLog ael = new EventLog("Application");
```

Comment:
EventLog is disposable. What this means is you should use a `using` statement whenever you have one, if you want it to be disposed at the end. Eg.,
```c#
using (EventLog ael = new EventLog("Application"))
{
 ... assert or whatever you do with ael ...
}
```

In cases where you actually created something on the machine, it may not be sufficient to dispose, you may haev to do something to delete it.